src/explore-education-statistics-admin/src/pages/release/data/utils/getApiDataSetFilterMappings.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  FilterCandidate,
+  FiltersMapping,
+  FilterMapping,
+  FilterColumnMapping,
+} from '@admin/services/apiDataSetVersionService';
+import { Dictionary } from '@common/types';
+
+export interface FilterCandidateWithKey extends FilterCandidate {
+  key: string;
+}
+export interface FilterMappingWithKey extends FilterMapping {
+  sourceKey: string;
+}
+
+export interface AutoMappedFilter {
+  candidate: FilterCandidateWithKey;
+  mapping: FilterMappingWithKey;
+}
+
+export interface MappableFilter {
+  candidate?: FilterCandidateWithKey;
+  mapping: FilterMappingWithKey;
+}
+
+export default function getApiDataSetFilterMappings(
+  filtersMapping: FiltersMapping,
+): {
+  autoMappedFilterOptions: Dictionary<AutoMappedFilter[]>;
+  newFilterOptionCandidates: Dictionary<FilterCandidateWithKey[]>;
```

Comment:
Minor, but maybe worth simplifying this name to `newFilterOptions`.

Similar could be done for `newFilterColumns` below as well.

Appropriate changes would also need to be made for any relevant variables in this function.